python/dataset.cpp

```diff
@@ -116,6 +116,7 @@ void init_dataset(py::module &m) {
       .def("__setitem__",
            [](Dataset &self, const std::string &name,
               const DataConstProxy &data) { self.setData(name, data); })
+      .def("__delitem__", &Dataset::erase)
```

Comment:
This should throw, to be consistent with a Python `dict`. I think this consistency is more important that the consistency with `std::map`, so I'd go with the simple/brief option of throwing in `erase`.

Reply:
:+1: